docs/tutorial/src/doc-00400.ts

```diff
@@ -81,52 +81,63 @@ export default async function doc_00400(cg: CandyGraph) {
-          colors: [0, 0, 0, 1],
-          widths: 17,
-        }),
-        createLineStrip(cg, xs, ys0, {
-          colors: [1, 0.5, 0, 1],
-          widths: 9,
-        }),
-        // Then our `ys1` trace in ~blue:
-        createLineStrip(cg, xs, ys1, {
-          colors: [0, 0, 0, 1],
-          widths: 17,
-        }),
-        createLineStrip(cg, xs, ys1, {
-          colors: [0, 0.5, 1, 1],
-          widths: 9,
-        }),
-      ]),
-      // Then we'll render our axes. Note that for the y-axis we're shifting the
-      // `axisIntercept` to keep up with the current time:
-      createOrthoAxis(cg, coords, "x", font, {
-        labelSide: 1,
-        tickLength: 5,
-        tickOffset: -2,
-        tickStep: 1,
-        labelFormatter: (n) => n.toFixed(0),
+
+    // First we'll prepare our plot data. We have two traces to render, `ys0`
+    // and `ys1`. We'll get a little fancy and render a black border around
+    // each trace by first rendering a thick black line, then a thinner line
+    // in our desired color. We'll also apply a scissor in screen space to
+    // confine the lines to the plot region. First the `ys0` trace in ~orange:
+    const lineStrips = [
+      createLineStrip(cg, xs, ys0, {
+        colors: [0, 0, 0, 1],
+        widths: 17,
+      }),
+      createLineStrip(cg, xs, ys0, {
+        colors: [1, 0.5, 0, 1],
+        widths: 9,
       }),
-      createOrthoAxis(cg, coords, "y", font, {
-        axisIntercept: time - history,
-        tickStep: 0.5,
-        tickLength: 5,
-        tickOffset: 2,
-        labelFormatter: (n) => n.toFixed(1),
+      // Then our `ys1` trace in ~blue:
+      createLineStrip(cg, xs, ys1, {
+        colors: [0, 0, 0, 1],
+        widths: 17,
       }),
-    ]);
+      createLineStrip(cg, xs, ys1, {
+        colors: [0, 0.5, 1, 1],
+        widths: 9,
+      }),
+    ];
+    const scissor = createScissor(cg, 40, 32, viewport.width - 56, viewport.height - 48, true, lineStrips);
+
+    // Then we'll prepare our axes. Note that for the y-axis we're shifting the
+    // `axisIntercept` to keep up with the current time:
+    const xaxis = createOrthoAxis(cg, coords, "x", font, {
+      labelSide: 1,
+      tickLength: 5,
+      tickOffset: -2,
+      tickStep: 1,
+      labelFormatter: (n) => n.toFixed(0),
+    });
+    const yaxis = createOrthoAxis(cg, coords, "y", font, {
+      axisIntercept: time - history,
+      tickStep: 0.5,
+      tickLength: 5,
+      tickOffset: 2,
+      labelFormatter: (n) => n.toFixed(1),
+    });
+
+    // Now we'll render the prepared data and axes:
+    cg.render(coords, viewport, [scissor, xaxis, yaxis]);
 
     // Finally, we'll copy our rendered plot to a canvas that's already been
     // added to this document:
     cg.copyTo(viewport, document.getElementById("doc_00400") as HTMLCanvasElement);
+
+    // At the end, we'll dispose of GPU resources, preventing memory leaks:
+    xaxis.dispose();
+    yaxis.dispose();
```

Comment:
Side-note. It would nice if we could skip disposing `yaxis`, because it doesn't change throughout the animation. It feels a bit wasteful to `.dispose()` it every frame. But it's better than a memory leak.